health-monitors/syslog-health-monitor/pkg/kata/detector.go

```diff
@@ -0,0 +1,545 @@
+		detectionTimeout: DefaultDetectionTimeout,
+		enableMetrics:    true, // Enable by default
+		rcCache: &runtimeClassCache{
+			ttl: DefaultRuntimeClassCacheTTL,
+		},
+	}
+
+	// Apply options
+	for _, opt := range opts {
+		opt(d)
+	}
+
+	return d, nil
+}
+
+// NewDetectorWithTimeout creates Kata detector with custom timeout.
+// Deprecated: Use NewDetector with WithTimeout option.
+func NewDetectorWithTimeout(nodeName string, clientset kubernetes.Interface, timeout time.Duration) *Detector {
+	detector, err := NewDetector(nodeName, clientset, WithTimeout(timeout))
+	if err != nil {
+		// For backward compatibility, log error but don't fail
+		slog.Warn("Failed to create detector with validation", "error", err, "node", nodeName)
+		// Return detector without validation (old behavior)
+		return &Detector{
+			nodeName:         nodeName,
+			clientset:        clientset,
+			detectionTimeout: timeout,
+			enableMetrics:    true,
+			rcCache: &runtimeClassCache{
+				ttl: DefaultRuntimeClassCacheTTL,
+			},
+		}
+	}
+
+	return detector
+}
+
+// IsKataEnabled determines if Kata Containers runtime is enabled on this node.
+// Uses concurrent Kubernetes API detection:
+// 1. Node status (ContainerRuntimeVersion, labels, annotations)
+// 2. RuntimeClass handlers
+//
+// Filesystem detection intentionally excluded - binaries/directories persist after
+// Kata is disabled, causing false positives. Only API reflects active config.
+//
+// Returns DetectionResult with attempted methods and any errors encountered.
+func (d *Detector) IsKataEnabled(ctx context.Context) (*DetectionResult, error) {
+	// Respect parent context cancellation
+	if err := d.checkContextCancellation(ctx); err != nil {
+		return nil, err
+	}
+
+	ctx, cancel := context.WithTimeout(ctx, d.detectionTimeout)
+	defer cancel()
+
+	result := &DetectionResult{
+		IsKata:           false,
+		Method:           DetectionMethodNone,
+		AttemptedMethods: make([]DetectionMethod, 0, 2),
+		Errors:           make([]error, 0, 2),
+	}
+
+	slog.Info("Detecting Kata Containers runtime via Kubernetes API", "node", d.nodeName)
+
+	// Validate prerequisites
+	if err := d.validateClientset(); err != nil {
+		return nil, err
+	}
+
+	// Fetch node once for both detection methods
+	node, err := d.getNode(ctx)
+	if err != nil {
+		return nil, fmt.Errorf("failed to get node: %w", err)
+	}
+
+	// Run concurrent detection methods
+	if err := d.runConcurrentDetection(ctx, cancel, node, result); err != nil {
+		return nil, err
+	}
+
+	// Record final result
+	d.recordDetectionResult(result)
+
+	return result, nil
+}
+
+// checkContextCancellation checks if the parent context is already cancelled
+func (d *Detector) checkContextCancellation(ctx context.Context) error {
+	select {
+	case <-ctx.Done():
+		return ctx.Err()
+	default:
+		return nil
+	}
+}
+
+// validateClientset ensures the Kubernetes clientset is available
+func (d *Detector) validateClientset() error {
+	if d.clientset == nil {
```

Comment:
Can we instead ensure the clientset is always passed? Maybe in main we exit with status code 1 if we can't get the clientset so we don't have to keep doing the nil check everywhere